src/server/enterprise/server/api_server.go

```diff
@@ -332,3 +348,228 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
+// ConfigMap and the '$(MODE)' reference is verbatim.
+func (a *apiServer) rollPachd(ctx context.Context, paused bool) error {
+	var (
+		kc        *kubernetes.Clientset = a.env.GetKubeClient()
+		namespace                       = a.env.Namespace
+	)
+	cc := kc.CoreV1().ConfigMaps(namespace)
+	c, err := cc.Get(ctx, "pachd-config", metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		// if the ConfigMap is not found, then the cluster is by definition unpaused
+		if !paused {
+			return nil
+		}
+		// Since pachd-config is not managed by Helm, it may not exist.
+		c = newPachdConfigMap(namespace)
+		c.Data["MODE"] = "paused"
+		if _, err := cc.Create(ctx, c, metav1.CreateOptions{}); err != nil {
+			return errors.Errorf("could not create configmap: %w", err)
+		}
+	} else if err != nil {
+		return errors.Errorf("could not get configmap pachd-config: %w", err)
+	} else {
+		// Curiously, Get can return no error and an empty configmap!
+		// If this happens, need to set the name and namespace.
+		if c.ObjectMeta.Name == "" {
+			c = newPachdConfigMap(namespace)
+		}
+		if c.Data == nil {
+			c.Data = make(map[string]string)
+		}
+		if paused && c.Data["MODE"] == "paused" {
+			// Short-circuit and do not update if configmap
+			// is already in the correct state.  This keeps
+			// the updatedAt semantics correct when checking
+			// the pause status.
+			return nil
+		} else if !paused && c.Data["MODE"] != "paused" {
+			// Short-circuit and do not update if configmap
+			// is already in the correct state.  This keeps
+			// the updatedAt semantics correct when checking
+			// the pause status.
+			return nil
+		}
+		if paused {
+			c.Data["MODE"] = "paused"
+		} else {
+			c.Data["MODE"] = a.env.UnpausedMode
+		}
+		if c.Annotations == nil {
+			c.Annotations = make(map[string]string)
+		}
+		c.Annotations[updatedAtFieldName] = time.Now().Format(time.RFC3339)
+		if _, err := cc.Update(ctx, c, metav1.UpdateOptions{}); err != nil {
+			return errors.Errorf("could not update configmap: %w", err)
+		}
+	}
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		dd := kc.AppsV1().Deployments(namespace)
+		d, err := dd.Get(ctx, "pachd", metav1.GetOptions{})
+		if err != nil {
+			return err //nolint:wrapcheck
+		}
+		// Updating the spec rolls the deployment, killing each pod and causing
+		// a new one to start.
+		d.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+		if _, err := dd.Update(ctx, d, metav1.UpdateOptions{}); err != nil {
+			return err //nolint:wrapcheck
+		}
+		return nil
+	}); err != nil {
+		return errors.Errorf("could not updated pachd deployment: %v", err)
+	}
+
+	return nil
+}
+
+func newPachdConfigMap(namespace string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pachd-config",
+			Namespace: namespace,
+		},
+		Data: map[string]string{"MODE": "full"},
```

Comment:
this should be set to env.UnpausedMode right?

Reply:
No, this is the argument passed to `--mode`, so it needs to be ∈{`full`, `paused`, `enterprise`, `sidecar`}.

Reply:
Weeell, it will get overwritten before being saved in every current code path, but … you are right, and I’ve fixed it so future code will be correct.